packages/react/menu/src/Menu.tsx

```diff
@@ -517,6 +501,13 @@ const MenuContentImpl = React.forwardRef<MenuContentImplElement, MenuContentImpl
       }
     };
 
+    React.useEffect(() => {
+      const handleKeyDown = () => (isKeyboardEntryRef.current = true);
+      const handlePointerDown = () => document.removeEventListener('keydown', handleKeyDown);
+      document.addEventListener('keydown', handleKeyDown, { once: true });
+      document.addEventListener('pointerdown', handlePointerDown, { once: true });
+    }, []);
```

Comment:
realised this is wrong because `onEntryFocus` fires when the content receives focus which isn't necessarily only when it first opens. 

will look into this.